File: Frontend/src/app/pages/model/pago.ts
import { Recibo } from "./recibo";

export interface Pago {
  id: number;
  paidAmount: string;
  balance: string;
  mora: string;
  fecha: string;
  comments: string;
  fecCreacion: string;
  usuCreacion: string;
  fecModifica: string;
  usuModifica: string;
  idRecibo: Recibo;
}

File: Backend/src/main/java/pe/com/model/Pago.java
package pe.com.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "pago")
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Pago {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column
    private String paidAmount;

    @Column
    private String balance;

    @Column
    private String mora;

    @Column
    private String fecha;

    @Column
    private String comments;

    @Column
    private String fecCreacion;

    @Column
    private String usuCreacion;

    @Column
    private String fecModifica;

    @Column
    private String usuModifica;

    @ManyToOne
    @JoinColumn(name = "id_recibo", nullable=false)
    private Recibo idRecibo;
}
